astropy/io/fits/tests/test_connect.py

```diff
@@ -74,6 +75,12 @@ def test_overwrite_with_path(self, tmp_path):
             t1.write(Path(filename), format="fits", overwrite=True)
         t1.write(Path(tmp_path / filename), format="fits", overwrite=True)
 
+    def test_write_to_fileobj(self):
+        # regression test for https://github.com/astropy/astropy/issues/17703
+        t = Table(self.data)
+        with TemporaryFile() as temp:
```

Comment:
Do we need to clean this file up?

Reply:
`On completion of the context or destruction of the file object the temporary file will be removed from the filesystem.` -- OK maybe not.